src/coreclr/jit/lsra.cpp

```diff
@@ -6756,7 +6756,7 @@ void LinearScan::resolveRegisters()
         assert(refPosIterator != refPositions.end());
         assert(currentRefPosition->refType == RefTypeBB);
         ++refPosIterator;
-        currentRefPosition = &refPosIterator;
+        currentRefPosition = refPosIterator != refPositions.end() ? &refPosIterator : nullptr;
 
         // Handle the RefPositions for the block
```

Comment:
The iterator section of the for statement below seems to have the same problem.

Also, there are multiple for looks with this pattern in this method.

Reply:
Boo. I do love the tricky comma operator.